### PR TITLE
Adding new verbose flag

### LINKED ISSUE #8809: Introduce verbose output
Helm currently has it's normal output and output with `--debug` messages. This is a proposal to introduce another level between normal and debug output.

kubectl, using klog, has the ability to set a verbosity level. This level is not clearly defined so users have to work it out. But, there are multiple levels. I wouldn't recommend we have that experience.

Homebrew, the mac package manager, has two levels. `--verbose` and `--debug` that can be layered on to the normal output. Verbose prints more information about what's happening while debug prints details you'd need if you want to debug something.

This feature request suggests the same idea for Helm. For example, if you used `--verbose` on `helm install` it would tell you what's happening as it locates the chart, downloads it, renders the templates, pushes them, etc. I wouldn't need to give lots of internal values and details that debug should supply.

Thoughts?

### PR DESCRIPTION
**Pull request by sorenmat**
**What this PR does / why we need it**:
This allow us to output the progress of the installation / upgrade
with a bit more detail.
This is particular useful in combination with the wait flag, as it
outputs the progress into the console.

**Special notes for your reviewer**:
Fixes: #8809 #7760 #2040 #8582 
Example output of `bin/helm upgrade --install my-kubeview kubeview/kubeview --version 0.1.19 --wait --verbose`                                                                                                         
```
Getting history for release my-kubeview
Release "my-kubeview" does not exist. Installing it now.
creating 5 resource(s)
Beginning wait for 5 resources with timeout of 5m0s
Service does not have load balancer ingress IP address: default/my-kubeview
Deployment is not ready: default/my-kubeview. 0 out of 1 expected pods are ready
Deployment is not ready: default/my-kubeview. 0 out of 1 expected pods are ready
Deployment is not ready: default/my-kubeview. 0 out of 1 expected pods are ready
Deployment is not ready: default/my-kubeview. 0 out of 1 expected pods are ready
NAME: my-kubeview
LAST DEPLOYED: Thu Feb  4 09:39:14 2021
NAMESPACE: default
STATUS: deployed
REVISION: 1
NOTES:
=====================================
==== KubeView has been deployed! ====
=====================================
  To get the external IP of your application, run the following:

  export SERVICE_IP=$(kubectl get svc --namespace default my-kubeview -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
  echo http://$SERVICE_IP

  NOTE: It may take a few minutes for the LoadBalancer IP to be available.
        You can watch the status of by running 'kubectl get --namespace default svc -w my-kubeview'
```
**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Comment by sorenmat** [comment]
@bacongobbler you seems to be involved in some of the discussions on the issues this addresses, any thoughts?

**Comment by bacongobbler** [comment]
In a recent Helm security audit, it was recommended by the auditors that a global `--log-level` flag should be introduced to the project.

I would be interested in PRs that introduce that behaviour at a higher level. The logging capabilities provided by languages like RUST_LOG seem like a more ambitious goal to achieve, but it will be worth it in the long run.

Reading through this PR it looks as though each package has to pass a `log.Logger` and consume that logger. There are also no way to separate between different levels of verbosity, which has been highly requested by many community members.

In other words, I'd aim to write a log interface that sits at a much higher level than what you've proposed here. Something akin to Rust's logging system would be much more pluggable and easier to maintain going forward.

**Comment by bacongobbler** [comment]
It is also worth keeping in mind that this clashes with the existing `-v` flag inherited from kubectl. `-v 10 --verbose` introduces two separate levels of logging. So the design here is really poor for the end user.

**Comment by sorenmat** [comment]
seems like this was a bad idea, according to the comments
